### PR TITLE
Allow custom operations to return a different class than the expected resource class

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2855, api-platform/api-platform#1172
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Comment by bendavies** [comment]
this obviously begs the question, what is `$strict` even for?

File: tests/Fixtures/TestBundle/Document/Payment.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     * @var int|null
+     *
+     * @ODM\Id(strategy="INCREMENT", type="integer")
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $amount;
```
Comment (teohhanhui) [inline tests/Fixtures/TestBundle/Document/Payment.php]: Not mapped as Doctrine MongoDB ODM does not have a decimal type. I'm not sure what's the right thing to do here... If I use `type="float"`, the type validation fails.

Comment (teohhanhui) [reply tests/Fixtures/TestBundle/Document/Payment.php]: https://github.com/doctrine/mongodb-odm/issues/1835

File: features/main/custom_operation.feature [hunk 2]
```diff
@@ -2,10 +2,13 @@ Feature: Custom operation
   As a client software developer
   I need to be able to create custom operations
 
+  Background:
```
Comment (soyuka) [inline features/main/custom_operation.feature]: oh this is nice, does this allow to add steps to every scenario?

Comment (teohhanhui) [reply features/main/custom_operation.feature]: Yeah.

**Comment by teohhanhui** [comment]
> Is the strict parameter still used?

No. But I don't think we should deprecate it either? There are tests that use it so it's at least covered.

**Comment by soyuka** [comment]
> No. But I don't think we should deprecate it either? There are tests that use it so it's at least covered.

Hmm yeah leave it like this, at first I thought that this class was `@internal`.

**Pull request merged**